src/serializationStrategies/XpathRangeSelector/xpath.ts

```diff
@@ -181,21 +181,29 @@ export function getFirstByXPath(path: string, offset: number, referenceElement:
   // for element targets, highlight children might be artifically
   // inflating the range offset, fix.
   if (node && isElement(node)) {
-    let search: Node | null = node.childNodes[offset - 1];
+    let search: Node | null = node.childNodes[0];
+    let offsetElementsFound = 0;
+    let modifyOffset = 0;
+
+    while (search && offsetElementsFound < offset) {
+      offsetElementsFound++;
 
-    while (search) {
       if (isTextOrTextHighlight(search)) {
         search = search.nextSibling;
 
         while (isTextOrTextHighlight(search!)) {
-          offset++;
+          modifyOffset++;
           search = search!.nextSibling;
         }
+      } else {
+        search = search ? search.nextSibling : null;
```

Comment:
This can be `search = search.nextSibling` like in line 192 because it is inside while loop which runs only if `search` exists